cmd/ci-operator/main_test.go

```diff
@@ -1612,3 +1614,109 @@ func TestHandleTargetAdditionalSuffix(t *testing.T) {
 		})
 	}
 }
+
+type mockResolverClient struct {
+	ClusterProfileFunc func(profileName string) (*api.ClusterProfileDetails, error)
+}
+
+func (m *mockResolverClient) ClusterProfile(profileName string) (*api.ClusterProfileDetails, error) {
+	if m.ClusterProfileFunc != nil {
+		return m.ClusterProfileFunc(profileName)
+	}
+	return nil, nil
+}
+
+func (m *mockResolverClient) Config(metadata *api.Metadata) (*api.ReleaseBuildConfiguration, error) {
+	return nil, nil
+}
+
+func (m *mockResolverClient) ConfigWithTest(base *api.Metadata, testSource *api.MetadataWithTest) (*api.ReleaseBuildConfiguration, error) {
+	return nil, nil
+}
+
+func (m *mockResolverClient) Resolve(data []byte) (*api.ReleaseBuildConfiguration, error) {
+	return nil, nil
+}
+
+func (m *mockResolverClient) IntegratedStream(namespace string, name string) (*configresolver.IntegratedStream, error) {
+	return nil, nil
+}
+
+func TestGetClusterProfileSecret(t *testing.T) {
```

Comment:
Technically you are not testing anything. The main operation that is being done to retrieve the cluster profiles is being tested in https://github.com/psalajova/ci-tools/blob/35c5f8f232ef977adf77ce20c237560df3e4eea8/pkg/registry/server/client_test.go#L88

Reply:
hm, right